wire-schema/src/test/java/com/squareup/wire/schema/PrunerTest.kt

```diff
@@ -1334,4 +1334,144 @@ class PrunerTest {
     assertThat(letterType.field("signature")!!.default).isEqualTo("Sent from Wire")
     assertThat(letterType.field("signature")!!.options.fields().isEmpty()).isTrue()
   }
+
+  @Test
+  fun sinceAndUntilRetainOlder() {
+    val schema = RepoBuilder()
+        .add("message.proto", """
+            |import "wire/extensions.proto";
+            |
+            |message Message {
+            |  optional string radio = 1 [wire.until=1950];
+            |  optional string video = 2 [wire.since=1950];
+            |}
+            """.trimMargin())
+        .schema()
+    val pruned = schema.prune(IdentifierSet.Builder()
+        .oldest(1949L)
+        .newest(1949L)
+        .build())
```

Comment:
will the pruner always run as part of wire codegen?  otherwise, i'd imagine this would also be useful in non-pruning scenarios.

Reply:
It’ll always run as a part of code gen. I think "pruning" is the implementation, whether it’s tree shaking or version stripping.